js/content/store.js

```diff
@@ -2662,53 +2660,41 @@ let FundsPageClass = (function(){
             );
         }
 
-        let currency = await Price.parseFromString(price, Currency.storeCurrency);
+        let currency = CurrencyRegistry.fromType(Currency.storeCurrency);
+        let minValue = currency.valueOf(price);
+        let symbol = currency.format.symbol;
+        let step = currency.format.decimalPlaces === 0 ? 1 : .01;
 
-        let inputel = newel.querySelector((giftcard ? "#es_custom_money_amount_wrapper" : ".price"));
-        HTML.inner(inputel, "<input type='number' id='es_custom_money_amount' class='es_text_input money' min='" + currency.value + "' step='.01' value='" + currency.value +"'>");
-        // TODO currency symbol
+        HTML.inner(
+            newel.querySelector(giftcard ? "#es_custom_money_amount_wrapper" : ".price"),
+            `${symbol}<input type="number" id="es_custom_money_amount" class="es_text_input money" min="${minValue}" step="${step}" value="${minValue}">`
+        );
 
-        document.querySelector((giftcard ? ".giftcard_selection" : ".addfunds_area_purchase_game"))
+        document.querySelector(giftcard ? ".giftcard_selection" : ".addfunds_area_purchase_game")
             .insertAdjacentElement("afterend", newel);
 
-        document.querySelector("#es_custom_money_amount").addEventListener("input", function() {
-            let value = document.querySelector("#es_custom_money_amount").value;
+        let btn = newel.querySelector("a.btn_medium");
 
-            if(!isNaN(value) && value != "") {
-                currency = new Price(value, Currency.storeCurrency);
-
-                if(giftcard) {
-                    priceel.classList.toggle("small", value > 10);
-                    priceel.textContent = currency;
-                }
+        document.querySelector("#es_custom_money_amount").addEventListener("input", e => {
+            let value = e.target.value;
+            if (value < minValue) {
+                value = minValue;
             }
-        });
 
-        newel.querySelector((giftcard ? ".es_custom_money a.btn_medium" : ".es_custom_button")).addEventListener("click", e => {
-            e.preventDefault();
-
-            let customAmount = Number(document.querySelector("#es_custom_money_amount").value).toFixed(2).replace(/[,.]/g, '');
+            let customAmount = Number(value).toFixed(2).replace(/[,.]/g, '');
 
             if (giftcard) {
-
-                if (e.target.closest(".giftcard_cont")) {
-                    ExtensionLayer.runInPageContext(amount => { submitSelectGiftCard(amount); }, [ customAmount ]);
-                }
-
+                priceel.classList.toggle("small", customAmount >= 100000); // 4 digits or longer
```

Comment:
I'd find it better if this would be based on characters, not digits.
For example 500,00€ is still in big font.

Reply:
ok, 8 characters seems to be about right